### DISCUSSION TITLE
When closing Audacity, the exe process still remains and unable to normally terminate the process

### DISCUSSION THREAD

**Discussion by Rocketboy235** [audacity/audacity#4676] [Q&A]
### Bug description

Audacity Version 3.2.5
Windows 10 Version 22H2 Build 19045.2846

Seems to be a rare bug but only local to my computer. Don't think anyone else is experiencing this? Tried uninstalling nd restarting the PC and then installing the program again but still get the same issue. Trying to figure out the best way to figure out the root cause. Also happened on a previous version of Audacity (but version 3, not 2).

Cannot use task manager. Access is denied.
Cannot use taskkill. Trying to do so will result in a response saying that there is no running instance of the task

### Steps to reproduce

May only be reproducible on my computer
1) Open Audacity
2) Open task manager and find the Audacity.exe process and monitor that
3) Close Audacity normally
4) Issue is shown

### Expected behavior

All audacity processes should terminate.

### Actual behavior

![image](https://user-images.githubusercontent.com/29405055/235309440-2991191a-4ae6-4bb3-9ae0-e28bb624ab7c.png)

After closing audacity, the process does not close and cannot be closed in task manager.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

**Comment by petersampsonaudacity** [comment]
I cannot reporoduce this on W10.

Looking at your image above I am very surprised gto see multiple instances of Audacity open, showing in Task Manager.

Normally I expect to only ever see a single instance in Task Manager even if I have multiple projects open.

![image](https://user-images.githubusercontent.com/50205984/235344519-d317a49e-6ca6-461f-b9c1-35956d2eeba8.png)

**Comment by Rocketboy235** [comment]
Yeah I think it’s probably just my OS that’s being glitchy. I only open it once and close that window but the process shows up 3x more after I do that strangely.

Earlier, when I closed Audacity, and then try to start the program again it would tell me that it’s already opened so I tried to kill it in task manager, which led to this ticket. But now it’s not much of an issue since I haven’t seen that pop up anymore and it’s not preventing me from opening it again. 

Not sure if you have any ways to debug this or a way to inspect some logs. I guess easiest solution is to reinstall Windows which I plan to do since there are other issues with it at the moment so may not be Audacity's fault.

**Comment by petersampsonaudacity** [comment]
You could try purging your Audacity settings folder:

See:  https://manual.audacityteam.org/man/installing_and_updating_audacity_on_windows.html#purge

**Comment by vsverchinsky** [comment]
Audacity may spawn a new process when testing new plugins, usually they either terminate normally or crash. If you could send a `pluginregistry.cfg` that might help to figure out where the problem is. `pluginregistry.cfg` may contain personal data like paths, but I need only names,  types and their state in the system. From what I see "Help->Diagnostics->Generate Support Data" doesn't not automatically strip sensitive information from the report either.

**Reply by Rocketboy235** [reply]
As of right now, only one audacity process spawns and will usually persist even after closing audacity. The process doesn't go away (at least the window closes) until I end the process in task manager and wait several minutes before it finally disappears from task manager.

**Comment by Rocketboy235** [comment]
Sorry for the extremely late reply. Was pretty busy and wasn't using Audacity much again until recently. Some things I noticed and maybe my reporting for the first time was not 100% accurate was that when attempting to kill the program in task manager, it seems to not complain the first time but the program still exists in task manager. Subsequent attempts to kill it in task manager will show the access is denied popup. What's interesting however is that as long as you attempted to end the program once, the process will eventually go away in like 5 minutes (maybe some sort of process timeout for zombie processes?)

I have attached the plguniregistry cfg file as requested. I don't think I have any third party plugins **enabled** that don't come with audacity though. There are a few disabled that audacity told me had issues and I was using spectralayers at one point but that is disabled.
[pluginregistry.cfg.txt](https://github.com/audacity/audacity/files/13700920/pluginregistry.cfg.txt)

I also did try resetting the settings but unfortunately original issue of 4 audacity stuck processes came back when I started the program up again. I decided to restore my settings to before so now its only showing 1 process.

Audacity 3.3.1
Windows 10 Version 22H2 Build 19045.3693

going to update to 3.4.2 but I don't think that will change anything (if it does, I'll report back, but assume it doesn't)

**Comment by Rocketboy235** [comment]
Seems like the issue went away after updating to v3.7.4, or maybe it went away after updating Windows and removing some problematic AV programs on my PC the same day earlier today 🤔 

Anyway, this can be closed now. Thank you 😄
